Source/Core/VideoCommon/HiresTextures.cpp

```diff
@@ -135,30 +188,111 @@ void HiresTexture::Update()
+               percent_mem_to_cull);
+    }
+  }
+
+  double prefetched_size_sum = 0;  // total mem size incl. non-native decompressed textures
+  const double absolute_max_mem = Common::MemPhysical();
+  // Currently "total usable", excl. HW reserved
+  const double culled_percent_amount = (Common::MemPhysical() * percent_mem_to_cull) / 100;
+  const double delegated_max_mem = Common::MemPhysical() - culled_percent_amount;
+  const double below_min_req_mem = 3221225472;  // 3 GB; 2GB Rec. + 1GB OS/Misc
+  // WIP - using 1024 * ... etc but makes the line so long, no global ref found, make one?
+
+  // message only
+  if (absolute_max_mem <= below_min_req_mem)
+  {
+    OSD::AddMessage("Prefetcher: Warning! System RAM of " +
+                        StringFromFormat("%.2f MB", absolute_max_mem / (1024.0 * 1024.0)) +
+                        " is below Dolphin's minimum recommended specifications.",
+                    20000);
+    WARN_LOG(VIDEO, "Usable system memory below Dolphin's minimum recommended specifications");
+  }
 
-  size_t size_sum = 0;
-  size_t sys_mem = Common::MemPhysical();
-  size_t recommended_min_mem = 2 * size_t(1024 * 1024 * 1024);
-  // keep 2GB memory for system stability if system RAM is 4GB+ - use half of memory in other cases
-  size_t max_mem =
-      (sys_mem / 2 < recommended_min_mem) ? (sys_mem / 2) : (sys_mem - recommended_min_mem);
-  u32 starttime = Common::Timer::GetTimeMs();
-  for (const auto& entry : s_textureMap)
+  // Routine Stat Notification
+  OSD::AddMessage("Prefetcher: Maximum delegated memory limit has been configured to " +
+                      StringFromFormat("%.2f MB", delegated_max_mem / (1024.0 * 1024.0)) +
+                      ", using cull of " + StringFromFormat("%.1f %%", percent_mem_to_cull),
+                  20000);
+
+  // Start Prefetching
+  prefetch_starttime_total = Common::Timer::GetTimeMs();
+
+  for (const auto& texmap_ref : s_textureMap)
   {
-    const std::string& base_filename = entry.first;
+    const std::string& base_filename = texmap_ref.first;
+    // const std::string& full_filepath = texmap_ref.second;
+    // idea to calc m_levels data_size per extension, but existing system uses base filenames
+    // would add ability to calc non-native texture(PNG) in mem and compare to ondisk size
+    // with the right formula that would return the amount of overhead
+    // (for 14 Megs of PNGs mem is ~350MB)
+    // to get the per-extension time probably this whole for loop would have to be redone
+    // I was not able to do this with full day hours while leaving "m_levels" unchanged
 
     if (base_filename.find("_mip") == std::string::npos)
     {
       {
         // try to get this mutex first, so the video thread is allow to get the real mutex faster
+        // JonnyH: so the prefetch() is running in a tight loop, and it's using that to allow
+        // "at least" 1 Search() run per cycle
+        // merry: exactly!
```

Comment:
After I talked about some unrelated thing about the mutex locks, Someone else (JonnyH i think) gone through and wasted 20 minutes on trying to figure out how these mutex locks 

I think this is a big enough "detail" to have it properly commented so not me or anyone else wastes 20 or more minutes figuring it out. Don't you think that is frustrating, people like me who are eager to do improvements and write code, I should be trying to avoid spending time looking at the code, it's all connected to the part of not being paid, every now and then it doesn't hurt to put a bit more effort in than necessary, I would not be even bringing that up, but across the board, if every piece of code takes so long to figure out it crosses a line, then it becomes a problem. I'm prepared to do more than necessary, but the requirement is it has to be as enjoyable as possible.